src/ui/Windows/ThisMfcApp.cpp

```diff
@@ -1002,6 +1005,55 @@ bool ThisMfcApp::ParseCommandLine(DboxMain &dbox, bool &allDone, bool &postMinim
   return true;
 }
 
+UINT ThisMfcApp::ResolveAllowScreenCaptureStateResourceId(UINT nIdFirst) const
+{
+  // nIdFirst is the first known valid state.
+  // nIdFirst - 1 should be a resource indicating a program/state error.
+  // From nIdFirst through to the last state (nIdFirst + MaxState - 1),
+  // the resource should reflect within itself the indicated state.
+  UINT nId = nIdFirst - 1;
+  if (m_allowScreenCaptureState >= Disallowed || m_allowScreenCaptureState < MaxState)
+    nId = nIdFirst + m_allowScreenCaptureState;
+  return nId;
+}
+
+CString ThisMfcApp::GetAllowScreenCaptureStateMessage(UINT nIdFirst) const
+{
+  UINT nId = ResolveAllowScreenCaptureStateResourceId(nIdFirst);
+  CString csExcludeOverridePhrase;
+  if (!csExcludeOverridePhrase.LoadStringW(nId))
+    csExcludeOverridePhrase = L"<error-invalid-state-message>";
+  return csExcludeOverridePhrase;
+}
+
+void ThisMfcApp::DetectImplicitAllowScreenCaptureOverride()
```

Comment:
> This really belongs in winutils.cpp. It should ideally just return a bool, indicating if we're running under console or not. If not (RDP/ICA), then all screen capture prevention logic should be silently disabled.

ACK on moving to WinUtils...

If we surface cause for pwsafe reducing security protections, we would need to return more than just a bool... in such a case, I'd return a pwsafe WinUtils enum indicating console, ica, rdp (and we can add as needed in the future). This would allow the app to get needed detail for tooltip messaging while not taking on WtsApi32.h as a dep elsewhere beyond WinUtils.

The one remaining item is whether you are firm on not expressing ica/rdp detail in tooltip messages re pwsafe's reduction in security in this area.

From other comments, I think you are considering a generic message agnostic of cause for allowing scr cap... let me suggest generally that detail that does not overcomplicate the message makes it more firmly connected with causal data which can have a number of benefits (i.e., tech support, catching discrepancies in reported system configurations and what messages indicate, helping a user more immediately glean cause specifics... perhaps catching discrepancies such as expecting RDP but seeing ICA).

The separate messages for each cause seem low cost for reasonable detail but either will work. What are your final thoughts on this?

Reply:
OK with returning an enum rather than a bool.
I see your points about specifying the protocol by name, but when  balancing the chance of a discrepancy between expected and detected against the clarity of the tooltip for a typical non-technical user, I still think that the text should not mention the protocol.